src/paths.rs

```diff
@@ -0,0 +1,91 @@
+use app_dirs::{AppDataType, AppInfo};
+use std::path::PathBuf;
+
+pub const LOCK_FILE_NAME: &str = "requirements.lock";
+
+const APP_INFO: AppInfo = AppInfo {
+    name: "dmenv",
+    author: "Tanker",
+};
+
+use crate::error::*;
+
+pub struct Paths {
+    pub project: PathBuf,
+    pub venv: PathBuf,
+    pub lock: PathBuf,
+    pub setup_py: PathBuf,
+}
+
+pub struct PathsResolver {
+    pub venv_outside_project: bool,
+    pub python_version: String,
+
+    pub project_path: PathBuf,
+}
+
+impl PathsResolver {
+    pub fn new(project_path: PathBuf, python_version: &str) -> Self {
+        PathsResolver {
+            venv_outside_project: false,
+            project_path,
+            python_version: python_version.into(),
+        }
+    }
+
+    pub fn paths(&self) -> Result<Paths, Error> {
+        let venv_path = self.get_venv_path()?;
+        Ok(Paths {
+            project: self.project_path.clone(),
+            venv: venv_path,
+            lock: self.project_path.join(LOCK_FILE_NAME),
+            setup_py: self.project_path.join("setup.py"),
+        })
+    }
+
+    fn get_venv_path(&self) -> Result<PathBuf, Error> {
+        if let Ok(existing_venv) = std::env::var("VIRTUAL_ENV") {
+            return Ok(PathBuf::from(existing_venv));
```

Comment:
Don't use `var` to read a path, always `var_os`. (Because paths on Linux are raw byte arrays, not unicode).
Also return is a little bit unidiomatic.

Alt:
```
if let Some(existing_venv) = std::env::var_os("VIRTUAL_ENV") {
    Ok(existing_venv.into())
} else if self.venv_outside_project {
    self.get_venv_path_outside()
} else {
    self.get_venv_path_inside()
}
```